beacon-chain/rpc/eth/shared/request.go

```diff
@@ -134,3 +137,36 @@ func IsOptimistic(
 	http2.WriteError(w, errJson)
 	return true, nil
 }
+
+// DecodeWithLength takes a string and a length and validates the hex while returning an error
+func DecodeWithLength(s string, length int) ([]byte, error) {
```

Comment:
I would rename it to `DecodeHexWithLength`. Otherwise the caller will have to go into the function to understand it.